From 50d5581f726d73690156d4d867920410b94bcbe7 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Sat, 10 May 2014 10:47:15 -0700
Subject: [PATCH 017/117] atomisp_gmin_platform: Create BYT clock device

The VLV2 clock output driver is a platform device and must be created.
Being an on-SoC device, it doesn't appear in the ACPI tables and can't
be probed that way.

This code is thus BYT-specific right now.  It needs a dynamic probe to
be added somehow, maybe via CPUID.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 arch/x86/platform/intel-mid/atomisp_gmin_platform.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
index 84b02a6..0820005 100644
--- a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
+++ b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
@@ -3,6 +3,7 @@
 #include <linux/dmi.h>
 #include <linux/efi.h>
 #include <linux/acpi.h>
+#include <linux/platform_device.h>
 #include <linux/atomisp_platform.h>
 #include <asm/spid.h>
 
@@ -202,3 +203,18 @@ int gmin_get_config_var(struct device *dev, const char *var, char *out, size_t *
 	return ret == EFI_SUCCESS ? 0 : -EINVAL;
 }
 EXPORT_SYMBOL_GPL(gmin_get_config_var);
+
+static int __init gmin_plat_init(void)
+{
+	/* BYT-T output clock driver required by the MIPI-CSI
+	 * camera modules */
+	if (IS_ERR(platform_device_register_simple("vlv2_plat_clk",
+						   -1, NULL, 0)))
+	{
+		pr_err("Failed to register vlv2_plat_clk device");
+		return -ENODEV;
+	}
+	return 0;
+}
+
+device_initcall(gmin_plat_init);
-- 
1.9.3

